sentry-core/src/main/java/io/sentry/core/MainEventProcessor.java

```diff
@@ -72,7 +72,14 @@ private void processNonCachedEvent(SentryEvent event) {
     }
 
     if (event.getThreads() == null) {
-      event.setThreads(sentryThreadFactory.getCurrentThreads());
+      // that means user has called Sentry.captureMessage
+      if (event.getMessage() != null && event.getExceptions() == null) {
+        if (options.isAttachStacktrace()) {
+          event.setThreads(sentryThreadFactory.getCurrentThreads());
+        }
+      } else {
+        event.setThreads(sentryThreadFactory.getCurrentThreads());
+      }
```

Comment:
I'd say that it's a good guess of detecting a `Sentry.captureMessage` call but it could be a custom `Sentry.captureEvent` call with message and exceptions and in this case, if `attachStacktrace` is `disabled`, it still reports threads.
what do you think @bruno-garcia